Slash/DB/MySQL/MySQL.pm

```diff
@@ -13695,6 +13698,43 @@ PRIMARY KEY (discussion_id, uid)
 		if (!$self->sqlDo("UPDATE users_comments SET commentsort = 1 WHERE commentsort = 5")) {
                         return 0;
                 }
+		#here
+		if (!$self->sqlDo("ALTER TABLE users_comments ADD mode_new ENUM('flat', 'nocomment', 'thread', 'threadtng', 'threadtos') NOT NULL DEFAULT 'threadtos'")) {
+			return 0;
+		}
+		if (!$self->sqlDo("UPDATE users_comments SET mode_new = mode")) {
+                        return 0;
+                }
+		if (!$self->sqlDo("ALTER TABLE users_comments DROP mode")) {
+                        return 0;
+                }
+		if (!$self->sqlDo("ALTER TABLE users_comments CHANGE mode_new mode ENUM('flat', 'nocomment', 'thread', 'threadtng', 'threadtos') NOT NULL DEFAULT 'threadtos'")) {
+                        return 0;
+                }
+		if (!$self->sqlDo("UPDATE users_comments set mode = 'threadtos' where mode = 'thread'")) {
+			return 0;
+		}
+		if (!$self->sqlDo("INSERT INTO commentmodes (mode, name) VALUES ('threadtos', 'Threaded-TOS')")) {
+			return 0;
+		}
+		if (!$self->sqlDo("INSERT INTO commentmodes (mode, name) VALUES ('threadtng', 'Threaded-TNG')")) {
+			return 0;
+		}
+		if (!$self->sqlDo("ALTER TABLE users_comments ADD mode_new ENUM('flat', 'nocomment', 'thread', 'threadtng', 'threadtos') NOT NULL DEFAULT 'threadtng'")) {
+                        return 0;
+                }
+                if (!$self->sqlDo("UPDATE users_comments SET mode_new = mode")) {
+                        return 0;
+                }
+                if (!$self->sqlDo("ALTER TABLE users_comments DROP mode")) {
+                        return 0;
+                }
+		if (!$self->sqlDo("ALTER TABLE users_comments CHANGE mode_new mode ENUM('flat', 'nocomment', 'threadtng', 'threadtos') NOT NULL DEFAULT 'threadtng'")) {
+                        return 0;
+                }
+		if (!$self->sqlDo("DELETE FROM commentmodes WHERE mode = 'thread'")) {
+			return 0;
+		}
```

Comment:
That's the safe and proper way to change the values of or add items to an enum in mysql. Create the new one, populate it from the old one, delete the old one, rename the new one.